src/java.base/share/classes/java/lang/invoke/VarHandles.java

```diff
@@ -744,48 +751,31 @@ private static boolean isCheckedException(Class<?> clazz) {
 //            System.out.println();
 //
 //            // Declare the stream of shapes
-//            Stream<HandleType> hts = Stream.of(
-//                    // Object->Object
-//                    new HandleType(Object.class, Object.class),
-//                    // Object->int
+//            List<HandleType> hts = List.of(
+//                    // Object->T
+//                    new HandleType(Object.class),
+//
+//                    // <static>->T
+//                    new HandleType(null),
+//
+//                    // Array[index]->T
 //                    new HandleType(Object.class, int.class),
-//                    // Object->long
+//
+//                    // MS[base]->T
 //                    new HandleType(Object.class, long.class),
-//                    // Object->float
-//                    new HandleType(Object.class, float.class),
-//                    // Object->double
-//                    new HandleType(Object.class, double.class),
-//
-//                    // <static>->Object
-//                    new HandleType(null, Object.class),
-//                    // <static>->int
-//                    new HandleType(null, int.class),
-//                    // <static>->long
-//                    new HandleType(null, long.class),
-//                    // <static>->float
-//                    new HandleType(null, float.class),
-//                    // <static>->double
-//                    new HandleType(null, double.class),
-//
-//                    // Array[int]->Object
-//                    new HandleType(Object.class, Object.class, int.class),
-//                    // Array[int]->int
-//                    new HandleType(Object.class, int.class, int.class),
-//                    // Array[int]->long
-//                    new HandleType(Object.class, long.class, int.class),
-//                    // Array[int]->float
-//                    new HandleType(Object.class, float.class, int.class),
-//                    // Array[int]->double
-//                    new HandleType(Object.class, double.class, int.class),
-//
-//                    // Array[long]->int
-//                    new HandleType(Object.class, int.class, long.class),
-//                    // Array[long]->long
+//
+//                    // MS[base][offset]->T
```

Comment:
I like how you programmatically generate all guards for all the shapes we care about.

On `MS[base][offset]` I wonder if generating a guard matters (for now) given that such a var handle will be indirect?

Reply:
Yes, it avoids the extra generation of vh linker lambda forms.